code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -754,7 +755,7 @@
 		logevent("Firmware [new_upgrade] run successfully.")
 		qdel(new_upgrade)
 		return FALSE
-	upgrades += new_upgrade
+	src.upgrades += new_upgrade
```

Comment:
Same here.

Reply:
upgrades is not accessible if src.upgrades is not explicitly declared in this case.